Source/Fuse.Elements/Resources/Exif/Exif.uno

```diff
@@ -0,0 +1,190 @@
+			foreach (var p in properties)
+			{
+				if (p.Id == OrientationTagId)
+				{
+					var buffer = new Buffer(p.Value);
+					var orientation = buffer.GetShort(0,true);
+					return orientation;
+				}
+			}
+			return 0;
+		}
+	}
+
+	[Require("Source.Include", "ImageIO/ImageIO.h")]
+	[Require("Source.Include", "CoreFoundation/CoreFoundation.h")]
+	[Require("Xcode.Framework", "ImageIO.framework")]
+	[Require("Xcode.Framework", "CoreFoundation.framework")]
+	extern(iOS) static class ExifIOSImpl
+	{
+		internal static ExifData FromByteArray(byte[] bytes)
+		{
+			var orientation = GetOrientation(bytes);
+			return new ExifData(orientation);
+		}
+
+		[Foreign(Language.ObjC)]
+		static int GetOrientation(byte[] bytes)
+		@{
+			CFDataRef data = CFDataCreateWithBytesNoCopy(NULL, (const UInt8 *)bytes.unoArray->Ptr(), bytes.unoArray->Length(), kCFAllocatorNull);
+			CGImageSourceRef imageSource = CGImageSourceCreateWithData(data, NULL);
+
+			NSArray *metadataArray = nil;
+
+			if (imageSource) {
+				CGImageMetadataRef metadata = CGImageSourceCopyMetadataAtIndex(imageSource, 0, NULL);
+				if (metadata) {
+					metadataArray = CFBridgingRelease(CGImageMetadataCopyTags(metadata));
+					CFRelease(metadata);
+				}
+				CFRelease(imageSource);
+			}
+
+			int rotation = 0;
+			NSString* tagTarget = @"Orientation";
+			if (metadataArray != nil)
+			{
+				for (id tag in metadataArray) {
+					CFStringRef tagName = CGImageMetadataTagCopyName((CGImageMetadataTagRef)tag);
+					if ([tagTarget isEqualToString: (__bridge NSString*)tagName]) {
+						CFTypeRef rot = CGImageMetadataTagCopyValue((CGImageMetadataTagRef)tag);
+						rotation = [((__bridge NSNumber*)rot) intValue];
+						break;
+					}
+				}
+			}
+
+			return rotation;
+
+		@}
+	}
+
+	[Require("Gradle.Dependency.Compile", "com.drewnoakes:metadata-extractor:2.10.1")]
+	[ForeignInclude(Language.Java,
+		"java.io.ByteArrayInputStream",
+		"java.io.InputStream",
+		"java.io.IOException",
+		"com.drew.imaging.ImageMetadataReader",
+		"com.drew.imaging.ImageProcessingException",
+		"com.drew.metadata.Metadata",
+		"com.drew.metadata.MetadataException",
+		"com.drew.metadata.exif.ExifIFD0Directory")]
+	//Uses https://drewnoakes.com/code/exif/
+	extern(Android) class ExifAndroidImpl
+	{
+		internal static ExifData FromByteArray(byte[] bytes)
+		{
+			var buf = ForeignDataView.Create(bytes);
+			var orientation = GetOrientation(buf);
+			return new ExifData(orientation);
+		}
+
+		[Foreign(Language.Java)]
+		static int GetOrientation(Java.Object buf)
+		@{
+			InputStream s = new com.fuse.android.ByteBufferInputStream((com.uno.UnoBackedByteBuffer)buf);
+			try {
+				Metadata metadata = ImageMetadataReader.readMetadata(s);
+				ExifIFD0Directory directory = metadata.getFirstDirectoryOfType(ExifIFD0Directory.class);
+
+				if (directory != null && directory.containsTag(ExifIFD0Directory.TAG_ORIENTATION)) {
+					return directory.getInt(ExifIFD0Directory.TAG_ORIENTATION);
```

Comment:
I think this still risks throwing in the case where `ExifIFD0Directory.TAG_ORIENTATION` isn't an int. But I guess that's such a bizarre case, that it's not worth worrying about. We'd just let it throw. In other words, this looks like a nice compromise.